src/DotVVM.Framework/Compilation/Javascript/JavascriptTranslatableMethodCollection.cs

```diff
@@ -176,6 +176,18 @@ public void AddDefaultMethodTranslators()
                     return JavascriptTranslationVisitor.TranslateViewModelProperty(args[0], (MemberInfo)dotvvmproperty.PropertyInfo ?? dotvvmproperty.PropertyType.GetTypeInfo(), name: dotvvmproperty.Name);
                 }
             ));
+
+            var whereMethod = typeof(Enumerable).GetMethods(BindingFlags.Public | BindingFlags.Static)
+                .Where(m => m.Name == "Where" && m.GetParameters().Length == 2 && m.GetParameters().Last().ParameterType.GetGenericTypeDefinition() == typeof(Func<,>)).Single();
+            AddMethodTranslator(whereMethod, translator: new GenericMethodCompiler(
+                args => args[1].Member("filter").WithAnnotation(ResultIsObservableArrayAnnotation.Instance)
+                    .Invoke(args[2]).WithAnnotation(ResultIsObservableAnnotation.Instance)));
+
+            var selectMethod = typeof(Enumerable).GetMethods(BindingFlags.Public | BindingFlags.Static)
+                .Where(m => m.Name == "Select" && m.GetParameters().Length == 2 && m.GetParameters().Last().ParameterType.GetGenericTypeDefinition() == typeof(Func<,>)).Single();
+            AddMethodTranslator(selectMethod, translator: new GenericMethodCompiler(
+                args => args[1].Member("map").WithAnnotation(ResultIsObservableArrayAnnotation.Instance)
+                    .Invoke(args[2]).WithAnnotation(ResultIsObservableAnnotation.Instance)));
```

Comment:
Why `.WithAnnotation(ResultIsObservableArrayAnnotation.Instance)`? The result is not obervable, or is it? If this is required for it to work, then there is something wrong in the JS translator.

Reply:
I also think that the result should not be observable. The created sample works also without the annotation.

Moreover, I added unit tests for the Javascript translation of `Where` and `Select` methods. Can you please check that the translations are correct? There is also a sample `LambdaExpressions` that you can try to play with these translations.